source/agora/consensus/data/genesis/package.d

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+
+    Contains primitives related to the genesis block
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.data.genesis;
+
+import agora.common.Amount;
+import agora.common.Hash;
+import agora.common.crypto.Key;
+import agora.consensus.data.Block;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.Transaction;
+
+/// Contains a pointer to the genesis block.
+version (unittest)
+{
+    import Mod = agora.consensus.data.genesis.Test;
+    private immutable(Block)* gen_block = &Mod.GenesisBlock;
+}
+else
+{
+    import Mod = agora.consensus.data.genesis.Coinnet;
+    private immutable(Block)* gen_block = &Mod.GenesisBlock;
+}
+
+/*******************************************************************************
+
+    Returns:
+        a reference to the genesis block.
+
+*******************************************************************************/
+
+pragma(inline, true)
+public ref immutable(Block) GenesisBlock () nothrow @safe @nogc
+{
+    return *gen_block;
+}
+
+/*******************************************************************************
+
+    Set a custom Genesis block. Any subsequent call to `GenesisBlock`
+    will refer to the Block as configured here.
+
+    Params:
+        block = the new Genesis block to use
+
+*******************************************************************************/
+
+pragma(inline, true)
+public void setGenesisBlock (immutable Block* block)
+    @safe
+{
+    import agora.consensus.validation.Block;
+
+    assert(block !is null);
+
+    if (auto reason = isGenesisBlockInvalidReason(*block))
+        throw new Exception(reason);
+
+    gen_block = block;
+}
+
+/*******************************************************************************
+
+    Returns:
+        a reference to the genesis transaction
+
+*******************************************************************************/
+
+pragma(inline, true)
+public ref immutable(Transaction) GenesisTransaction () nothrow @safe @nogc
+{
+    return gen_block.txs[1];
+}
+
+
+/*******************************************************************************
+
+    Build a Genesis block from the provided data
+
+    This function is intended to be used when the hard-coded genesis block
+    needs to be replaced, e.g. when the parameters are changed or a new network
+    is created. It shouldn't be used in production code.
+
+*******************************************************************************/
+
+public immutable(Block) makeGenesis (
```

Comment:
As mentioned in the comment, this is here so that changing the genesis block is not such as hassle as it is now.
I might move it to a `utils` module at some point.

Reply:
Ahh ok.